enumerable.rb

```diff
@@ -0,0 +1,182 @@
+# CyclomaticComplexity Bug
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/MethodLength,Metrics/ModuleLength
+module Enumerable
+  def my_each
+    array_enum = is_a?(Range) ? to_a : self
+    result = ''
+    i = 0
+    if block_given?
+      while i < array_enum.size
+        yield(array_enum[i])
+        i += 1
+      end
+    else
+      result = array_enum.to_enum
+    end
+    !result.is_a?(Enumerator) ? array_enum : result
+  end
+
+  def my_each_with_index
+    array_enum = is_a?(Range) ? to_a : self
+    i = 0
+    result = ''
+    if block_given?
+      while i < array_enum.size
+        yield(array_enum[i], i)
+        i += 1
+      end
+    else
+      result = array_enum.to_enum :each_with_index
+    end
+    !result.is_a?(Enumerator) ? array_enum : result
+  end
+
+  def my_select
+    array_enum = is_a?(Range) ? to_a : self
+    res_array = []
+    result = ''
+    if block_given?
+      array_enum.my_each do |item|
+        res_array << item if yield(item)
+      end
+    else
+      result = array_enum.to_enum :select
+    end
+    !result.is_a?(Enumerator) ? array_enum : result
+  end
+
+  def my_all?(*args, &block)
+    array_enum = is_a?(Range) ? to_a : self
+    flag = true
+    if array_enum.empty?
+      flag = true
+    elsif !args[0].nil? && args[0].class == Class
+      array_enum.my_each { |item| flag = false unless item.is_a?(args[0]) }
+    elsif !args[0].nil?
+      if args[0].is_a?(Regexp)
+        string_regex = array_enum.join(' ')
+        flag = false unless string_regex =~ args[0]
+      else
+        array_enum.my_each { |item| flag = false unless item == args[0] }
+      end
+    elsif !block.nil?
+      array_enum.my_each { |item| flag = false unless block.call(item) }
+    else
+      array_enum.my_each { |item| flag = false unless item }
+    end
+    flag
+  end
```

Comment:
my_all? - when a Regex is passed as an argument returns true if all of the collection matches the Regex.

`%w[ant bear cat].all?(/t/) ` should return false
`%w[ant tiger cat].all?(/t/) ` should return true